contract/src/main/kotlin/io/provenance/scope/loan/utility/DataExtensions.kt

```diff
@@ -1,22 +1,32 @@
 package io.provenance.scope.loan.utility
 
-import com.google.protobuf.Message
-import com.google.protobuf.Timestamp
-import io.provenance.scope.util.toInstant
-import tech.figure.util.v1beta1.Checksum
-import tech.figure.util.v1beta1.Date
-import tech.figure.util.v1beta1.Money
-import tech.figure.util.v1beta1.UUID
-import java.time.Instant
+import com.google.protobuf.Message as ProtobufMessage
+import com.google.protobuf.Timestamp as ProtobufTimestamp
+import java.util.UUID as JavaUUID
+import tech.figure.util.v1beta1.Checksum as FigureTechChecksum
+import tech.figure.util.v1beta1.Date as FigureTechDate
+import tech.figure.util.v1beta1.Money as FigureTechMoney
+import tech.figure.util.v1beta1.UUID as FigureTechUUID
 
-internal fun Message?.isSet() = this !== null && this != this.defaultInstanceForType
+@Suppress("TooGenericExceptionCaught")
+private fun tryOrFalse(fn: () -> Any): Boolean =
+    try {
+        fn()
+        true
+    } catch (e: Exception) {
+        false
+    }
 
-internal fun Timestamp?.isValid() = isSet() && this!!.toInstant() < Instant.now()
+internal fun ProtobufMessage?.isSet() = this !== null && this != this.defaultInstanceForType
 
-internal fun Date?.isValid() = isSet() && this!!.value.isNotBlank()
+internal fun ProtobufTimestamp?.isValid() = isSet()
```

Comment:
I removed the extra checks from `Money.isValid()` and `Timestamp.isValid` to avoid making specific and unjustified assumptions about data. They are left defined despite now being equivalent to `isSet()` for clarity.